controlplane/kubeadm/controllers/controller.go

```diff
@@ -227,6 +227,13 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		// Create new Machine w/ init
 		logger.Info("Initializing control plane", "Desired", desiredReplicas, "Existing", numMachines)
 		return r.initializeControlPlane(ctx, cluster, kcp, controlPlane)
+	// Perform a general health check on control plane before continue
+	case true:
+		result, err := r.generalHealthCheck(ctx, cluster, kcp, controlPlane)
+		if err != nil {
+			return result, err
+		}
+		fallthrough
```

Comment:
Can we split the switch case instead of using `fallthrough` and a case that will always have true?